src/Extend/Auth.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Illuminate\Contracts\Container\Container;
+
+class Auth implements ExtenderInterface
+{
+    private $addPasswordCheckers = [];
+    private $removePasswordCheckers = [];
+
+    /**
+     * Add a new password checker.
+     *
+     * @param string $identifier: Unique identifier for password checker.
+     * @param string $passwordChecker: A closure or invokable class that contains the logic of the password checker.
```

Comment:
Needs `callable|string` docblock

Reply:
For the hasher, couldn't an extension just override `$container->make('hash')` via a ServiceProvider extender?